internal/services/oracle/oracle_autonomous_database_cross_region_disaster_recovery_resource.go

```diff
@@ -0,0 +1,350 @@
+			Computed: true,
+		},
+
+		"tags": commonschema.TagsForceNew(),
+
+		//computed
+		"remote_disaster_recovery_type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+
+		"auto_scaling_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+			Optional: true,
+		},
+		"auto_scaling_for_storage_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+			Optional: true,
+		},
+		"backup_retention_period_in_days": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+			Optional: true,
+		},
+		"character_set": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"compute_count": {
+			Type:     pluginsdk.TypeFloat,
+			Computed: true,
+			Optional: true,
+		},
+		"compute_model": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"data_storage_size_in_tb": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+			Optional: true,
+		},
+		"database_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"database_workload": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"license_model": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"mtls_connection_required": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+			Optional: true,
+		},
+		"national_character_set": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"customer_contacts": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Optional: true,
+			Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+		},
+	}
+}
+
+func (AutonomousDatabaseCrossRegionDisasterRecoveryResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (AutonomousDatabaseCrossRegionDisasterRecoveryResource) ModelObject() interface{} {
+	return &AutonomousDatabaseCrossRegionDisasterRecoveryResourceModel{}
+}
+
+func (AutonomousDatabaseCrossRegionDisasterRecoveryResource) ResourceType() string {
+	return "azurerm_oracle_autonomous_database_cross_region_disaster_recovery"
+}
+
+func (r AutonomousDatabaseCrossRegionDisasterRecoveryResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model AutonomousDatabaseCrossRegionDisasterRecoveryResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			sourceId, err := autonomousdatabases.ParseAutonomousDatabaseID(model.SourceAutonomousDatabaseId)
+			if err != nil {
+				return err
+			}
+			sourceDb, err := client.Get(ctx, *sourceId)
+			if err != nil {
+				return fmt.Errorf("fetching source DB: %+v", err)
+			}
```

Comment:
For consistency could you update this err msg to:

```suggestion
			if err != nil {
				return fmt.Errorf("retrieving %s: %+v", sourceId, err)
			}
```